share/python/procedures/findif_response_utils/tensor_helper.py

```diff
@@ -0,0 +1,118 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2016 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+"""
+Module of helper functions for distributed ccresponse computations.
+
+Defines functions for retrieving tensors computed at displaced geometries.
+"""
+from __future__ import absolute_import
+from __future__ import print_function
+import collections
+import shelve
+import copy
+import os
+import psi4
+import p4util
+from p4const import *
+
+
+def synthesize_displaced_tensor(db, signature, row_dim):
```

Comment:
@dgasmith, thanks for your comments.

re
I actually set up this reorganization because we're starting on work on additional findif operations.  I would want to avoid using `roa` in the naming within the module, so that I don't have to change them again with the addition of our current work.  The goal of the PR was to make these functions, previously part of `roa.py`, available for use in the new operations we are working on. I adapted these function from the old roa versions with the goal of removing anything that made them specific to `run_roa()`. 

The new operations should be ready shortly if that matters, but if you would prefer I change the module name to be roa specific for now I am fine with that. 

The `synthesize_displaced_tensor` function is constructing a list of the "flattened" tensors at each displaced geometry. My only goal in naming these functions was to be descriptive.  If you would like me to avoid using "tensor" I think `collect_displaced_matrix_data` would be descriptive while also avoiding confusion over "tensor".  

Similarly, for `grab_psi4_matrix` function I would propose `parse_geometry_matrix_data` as an alternative name. Using matrix_data over matrix would emphasize that the return in a list of values, not a list of lists, or other matrix-like structure. I also agree the the name of the function should somehow indicate it is specific to 3*n matrices, and I think adding `geometry` would be a fine way to do that.  

 Would these changes to the naming be satisfactory?